src/app/home/home.component.ts

```diff
@@ -167,4 +176,128 @@ export class HomeComponent implements OnInit, AfterViewInit, OnDestroy {
     },  (err) => console.log(err));
   }
 
+  onSubmit(credentialData, userDetail) {

```

Comment:
These has been copied from users profile which is duplicated. So I think we need to make it as a service.

Reply:
I think a directive would be better, but I agree we should consolidate this code into one file